### PR TITLE
Message class: deprecations and new methods

### PR DESCRIPTION
**Pull request by wagner-certat**
* deprecates Message.contains
* deprecate force parameter in Message.add and add new overwrite parameter
* adds Message.is_valid
  (Note: I couldn't reduce the duplicate code in .add and .is_valid here, because some values are ignored while all other are explicitly invalid.)
* Added the raise_failure parameter to Message.add

Please review and give feedback if this is ok. Then I will do the replacements in the code base.

### CODE REVIEW

File: intelmq/lib/message.py [hunk 1]
```diff
@@ -106,14 +106,47 @@ def __init__(self, message=(), auto=False, harmonization=None):
     def __setitem__(self, key, value):
         self.add(key, value)
 
-    def add(self, key, value, sanitize=True, force=False, ignore=()):
-        if not force and key in self:
-            raise exceptions.KeyExists(key)
+    def is_valid(self, key, value, sanitize=True):
+        """
+        Checks if a value is valid for the key.
```
Comment (SYNchroACK) [inline intelmq/lib/message.py]: Add a phrase regarding `sanitize` parameter as a dev documentation.

Comment (SYNchroACK) [inline intelmq/lib/message.py]: **Suggestion:** Add a phrase regarding sanitize parameter as a dev documentation.

File: intelmq/lib/message.py [hunk 2]
```diff
@@ -106,14 +106,47 @@ def __init__(self, message=(), auto=False, harmonization=None):
...
+        if sanitize:
+            value = self.__sanitize_value(key, value)
+        valid = self.__is_valid_value(key, value)
+        if valid[0]:
+            return True
+        return False
 
-        if value in ["-", "N/A"]:
+    def add(self, key, value, sanitize=True, force=False, overwrite=False, ignore=(),
+            raise_failure=True):
```
Comment (SYNchroACK) [inline intelmq/lib/message.py]: **Suggestion:** Add the same kind of documentation as you have in `is_valid` method with special attention to `raise_failure` parameter.

**Comment by wagner-certat** [comment]
> Should we replace all bots code which is using the is_valid method from each harmonization type (String, FQDN, URL, etc)?

Yes, I will do so. If the library changes are ok.

File: intelmq/bots/experts/abusix/expert.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def process(self):
             abuse_contact_key = key + "abuse_contact"
             if abuse_contact_key in event and not self.parameters.overwrite:
                 continue
-            if event.contains(ip_key):
+            if ip_key in event:
```
Comment (aaronkaplan) [inline intelmq/bots/experts/abusix/expert.py]: I really prefer this syntax: if ip_key in event:
That's may more pretty.

**Comment by SYNchroACK** [comment]
> Yes, I will do so. If the library changes are ok.

Yap, everything ok @wagner-certat

**Comment by wagner-certat** [comment]
I did all replacements, please have a quick glance.

**Comment by SYNchroACK** [comment]
> I did all replacements, please have a quick glance.

Good to go! Nice work

**Pull request merged**